front-end/src/components/CustomInput.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useRef, useState } from "react"
+import { Markdown } from "@/components/Markdown.tsx"
+import { Input } from "@/components/ui/input"
+import { useFormContext } from "@/hooks/useFormContext.ts"
+import { useTranslation } from "../hooks/useTranslation"
+
+export const CustomInput: React.FC<{ id: string }> = ({ id }: { id: string }) => {
+  const useInputContext = useFormContext()
+
+  const inputSplit = id.split("#")
+  const inputID = inputSplit[0]
+
+  // case over -> overlay | below -> no overlay, move other stuff down
+
+  const {
+    feedbackVariation,
+    first,
+    feedback,
+    align,
+    placeholder,
+    prompt,
+    modeID,
+    questionMode,
+    setText,
+    text,
+  } = useInputContext[inputID]
+
+  // To select the first created input field on the site
+  const firstInputRef = useRef<HTMLInputElement | null>(null)
+  useEffect(() => {
+    if (firstInputRef.current) {
+      firstInputRef.current.focus()
+    }
+  }, [inputID])
+
+  const inputBorderColor = modeID === "invalid" ? "border-red-500 focus:border-red-500" : ""
+
+  const alignDiv: string = "w-full h-full"
+
+  let promptElement
+  if (prompt) {
+    promptElement = <Markdown md={prompt} />
+  }
+
+  let spacing
+  if (align === "NL") {
+    spacing = <br />
+  }
+
+  const [isInputFocused, setIsInputFocused] = useState(false)
+  let inputElement
+  // constantly below the input field
+  if (feedbackVariation === "below") {
+    inputElement = (
+      <div>
+        {spacing}
+        <div className="flex flex-col">
+          <div className="flex flex-row items-center">
+            <div className="mr-2">{promptElement}</div>
+            <div className={`${alignDiv} relative`}>
+              <Input
+                ref={first ? firstInputRef : null}
+                key={`newline-input-${inputID}`}
+                autoFocus
+                disabled={questionMode === "correct" || questionMode === "incorrect"}
+                value={text || ""}
+                onChange={(e) => {
+                  setText ? setText(e.target.value) : ""
+                }}
+                type="text"
+                className={`${inputBorderColor} focus:outline-none`}
+                placeholder={placeholder || ""}
+              />
+            </div>
+          </div>
+          <FeedbackComponent
+            formatFeedback={feedback ? feedback : ""}
+            mode={modeID ? modeID : "draft"}
+            type={"below"}
+          />
+        </div>
+        {spacing}
+      </div>
+    )
+  } else {
+    inputElement = (
+      <div>
+        {spacing}
+        <div className="flex flex-row items-center">
+          <div className="mr-2">{promptElement}</div>
```

Comment:
Der Prompt sollte meines Erachtens einfach im Markdown-Text stehen, damit auch Lückentexte etc. möglich sind. Wenn man es in der Tabelle haben will, nutzt man Tabellen-Notation:

| --- | --- |
| Prompt | Eingabfeld |
| Fragentext1 | {{! inputfeld1#bla#blubb !}} |
| Fragentext2 | {{! inputfeld2#bla#blubb !}} |

Dadurch wird der Prompt in Markdown gerendert. Nur das Eingabefeld muss hier in aktuell CustomInput erzeugt werden, nicht der Prompt. Oder übersehe ich etwas?

Reply:
Ich weiß was du meinst und ich hatte es auch schon anders implementiert. 

Die Darstellung in einer Tabelle funktioniert (nutze ich aktuell auch bei der Stack und Queue Frage). 
Ich würde den Prompt weiterhin so lassen wie er aktuell ist, da sich dadurch das Feedback, das InputFeld und der Prompt besser aneinander ausrichten (diese sind dann ein zusammenhängender Block). 

Lückentexte werden noch nicht unterstützt. Ich habe es nicht hinbekommen Feedback darunter anzuzeigen, welches zentriert unter dem Feld ist, aber dennoch größer ist als das Feld. Da mir das nicht gefallen hat habe ich es rausgenommen.

Reply:
Soll ich das noch einfügen oder wollen wir das einfügen, wenn uns Bedarf einfällt?